### PR TITLE
Move binfmt to a multi-stage build

### PR DESCRIPTION
**Pull request by rneugeba**
also update all YAML files to the new image

### CODE REVIEW

File: pkg/binfmt/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+    echo "golint..." && test -z $(golint main.go)
+RUN go build -o usr/bin/binfmt -buildmode pie -ldflags "-s -w -extldflags -static" main.go
+
 FROM scratch
-COPY . ./
+ENTRYPOINT []
+WORKDIR /
+COPY --from=qemu-build usr/bin/qemu-* usr/bin/
+COPY --from=binfmt-build usr/bin/binfmt usr/bin/binfmt
+COPY etc/binfmt.d/00_linuxkit.conf etc/binfmt.d/00_linuxkit.conf
```
Comment (AkihiroSuda) [inline pkg/binfmt/Dockerfile]: unneeded?

File: pkg/binfmt/Dockerfile [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+FROM linuxkit/go-compile:4513068d9a7e919e4ec42e2d7ee879ff5b95b7f5@sha256:bdfadbe3e4ec699ca45b67453662321ec270f2d1a1dbdbf09625776d3ebd68c5 AS binfmt-build
+COPY main.go main.go
+RUN echo "gofmt..." && test -z $(gofmt -s -l main.go) && \
+    echo "go vet..." && test -z $(GOOS=linux go tool vet -printf=false main.go) && \
+    echo "golint..." && test -z $(golint main.go)
+RUN go build -o usr/bin/binfmt -buildmode pie -ldflags "-s -w -extldflags -static" main.go
+
 FROM scratch
-COPY . ./
+ENTRYPOINT []
```
Comment (AkihiroSuda) [inline pkg/binfmt/Dockerfile]: unneeded?

File: pkg/binfmt/Dockerfile [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+WORKDIR /
```
Comment (AkihiroSuda) [inline pkg/binfmt/Dockerfile]: unneeded?

File: pkg/binfmt/Dockerfile (hunk 2 above)
Comment (rneugeba) [inline pkg/binfmt/Dockerfile]: the go-compile images used in the second stage defines an entry point and on a previous conversion i had issues with entrypoints/workdir/cmd getting inherited, so started the habit of resetting them in the final layer just to make sure

**Comment by justincormack** [comment]
Looks fine for now, makes it easier to continue to improve the package builds...

**Pull request merged**